include/xtensor/xsort.hpp

```diff
@@ -0,0 +1,411 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay and Wolf Vollprecht    *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_SORT_HPP
+#define XTENSOR_SORT_HPP
+
+#include <iterator>
+#include <algorithm>
+
+#include "xarray.hpp"
+#include "xeval.hpp"
+#include "xtensor.hpp"
+#include "xslice.hpp"  // for xnone
+
+namespace xt
+{
+    namespace detail
+    {
+
+        template <class T>
+        class strided_iterator :
+            public std::iterator<std::random_access_iterator_tag, T, ptrdiff_t, const T*, T&>
```

Comment:
I don't understand the need for this. Why not using steppers ? (they basically do the same and work with any kind of expession, not only the strided ones).

I think this deserves a dedicated PR.

Reply:
Is there a way to give the stepper a RandomIterator interface (e.g. `++` and `+(int)`)? Instead of `step(steps, dim)`?
I'd love to remove all this code!

Reply:
Let's discuss this in the PR dedicated to `xsort.hpp`.